Allow multiple --network flags for podman run/create

Luap99 opened the pull request:
We allow a container to be connected to several cni networks
but only if they are listed comma sperated. This is not intuitive
for users especially since the flag parsing allows multiple string
flags but only would take the last value. see: spf13/pflag#72

Also get rid of the extra parsing logic for pods. The invalid options
are already handled by `pkg/specgen`.

A test is added to prevent a future regression.

<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/master/CONTRIBUTING.md#submitting-pull-requests).

In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]".  That will prevent functional tests from running and save time and energy.
-->

mheon commented:
I'm not strictly opposed, but if we do this we need to be very sure the manpage makes it clear that this is only for multiple networks - I'm afraid of users thinking that `podman run --net slirp4netns --net host` and similar invocations are valid.

rhatdan commented:
Could we add a failure mode where the user did do `--net=host --net=container:foobar`

TomSweeneyRedHat commented:
The code LGTM, but share @mheon's thought on man page updates for this.

zhangguanzhang commented:
> Could we add a failure mode where the user did do `--net=host --net=container:foobar`

agree

Luap99 commented:
Updated. This is now the error.
```
$ podman run --network host --network foo-a alpine
Error: network conflict between type host and bridge
```
I also removed the extra pod parsing logic. The invalid options are already handled in `pkg/specgen`

mheon commented:
/approve
Restarted one test which looked flaked.
LGTM

TomSweeneyRedHat commented on docs/source/markdown/podman-create.1.md:
```diff
@@ -584,7 +584,7 @@ Valid _mode_ values are:
 - **container:**_id_: reuse another container's network stack;
 - **host**: use the Podman host network stack. Note: the host mode gives the container full access to local system services such as D-bus and is therefore considered insecure;
-- _network-id_: connect to a user-defined network, multiple networks should be comma separated;
+- **cni-network**: connect to a user-defined network, multiple networks should be comma separated or can specified with multiple uses of **--network**;
```
```suggestion
- **cni-network**: connect to a user-defined network, multiple networks should be comma-separated or they can be specified with multiple uses of the **--network** option;
```

TomSweeneyRedHat commented on docs/source/markdown/podman-run.1.md:
```diff
@@ -610,7 +610,7 @@ Valid _mode_ values are:
 - **container:**_id_: reuse another container's network stack;
 - **host**: use the Podman host network stack. Note: the host mode gives the container full access to local system services such as D-bus and is therefore considered insecure;
-- _network-id_: connect to a user-defined network, multiple networks should be comma separated;
+- **cni-network**: connect to a user-defined network, multiple networks should be comma separated or can specified with multiple uses of **--network**;
```
```suggestion
- **cni-network**: connect to a user-defined network, multiple networks should be comma-separated or they can be specified with multiple uses of the **--network** option;
```

Luap99 commented:
Tests are green.
@containers/podman-maintainers PTAL

TomSweeneyRedHat commented:
LGTM
Thanks @Luap99 !

mheon commented:
/lgtm
